### PR TITLE
[MRG] Support cross-talk and fine-calibration files

### LINKED ISSUE #495: MEG fine-calibration and crosstalk files
So it seems the BIDS MEG specification does not cover MEG fine-calibration and crosstalk files.

> Note that we do not provide specifications for cross-talk and fine-calibration matrix files in the current version of BIDS.

https://bids-specification.readthedocs.io/en/stable/99-appendices/06-meg-file-formats.html#neuromagelektamegin

This is … let's just say **unfortunate**, as typically we require these files to run Maxwell filters or Elekta's MaxFilter on the data to even make them **usable**. Without these files, the recordings are basically trash. It is therefore **of uttermost importance** to distribute these files along with the recorded experimental data.

Now it has been [suggested](https://github.com/bids-standard/bids-examples/issues/56#issuecomment-372744313) to place these files in the `derivatives/` folder, [as done for the ds000248 OpenNeuro dataset](https://github.com/OpenNeuroDatasets/ds000248/tree/master/derivatives). 

However, these files can **not** be created from the raw recordings. Therefore, they are actually **not** derivatives in the BIDS sense.

Further, since derivatives are not yet fully specified, there is no agreed-upon naming scheme, making fully automated processing of the data impossible.

So to me it's clear: this needs to become part of the standard ASAP.

But "ASAP" in BIDS Time can mean "quite long" in Human Time.

Therefore I would like to propose to

1. start pushing forward a proposal to amend the BIDS standard (which will take ~long to complete);and while this is cooking, we already
1. urge ourselves to decide upon a storage / naming scheme for fine-calibration and crosstalk files **within MNE-BIDS** – producing something usable even **before the standard is finally amended**.

Users could leave it to MNE-BIDS to store those files for them properly, even if that would be "outside the official standard" in the beginning.

My idea is that `write_raw_bids()` would gain two keyword arguments, `fine_calib_fname` and `crosstalk_fname`, which allow specification of those files; and they would be placed in the appropriate (decided by us) places in the BIDS hierarchy.

Once 1. is finally finished, we adopt the then-standardized storage method. We could even offer a convenience function to convert from "our" storage (2.) to the standard storage (1.) automatically.

Thoughts?

### PR DESCRIPTION
**Pull request by hoechenberger**
PR Description
--------------

Adds support for Elekta/Neuromag/MEGIN cross-talk and fine-calibration files. Closes #495.

- `mne_bids.write_cross_talk()`
- `mne_bids.write_fine_calibration()`
- `BIDSPath.cross_talk_fpath` property*
- `BIDSPath.fine_calibration_fpath` property*
- CLI: `mne_bids crosstalk_to_bids`
- CLI: `mne_bids calibration_to_bids`

BIDS spec:
https://bids-specification.readthedocs.io/en/latest/99-appendices/06-meg-file-formats.html#neuromagelektamegin

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by hoechenberger** [comment]
This is currently blocked by an upstream PR for bids-validator, https://github.com/bids-standard/bids-validator/pull/1079

File: mne_bids/tests/test_write.py
Comment (sappelhoff) [inline mne_bids/tests/test_write.py:1631]: can you add a test what happens when a dict is passed? --> (1) a valid dict as read from the mne.preprocessing func, and (2) an arbitrary dict (which should raise an error)

File: mne_bids/tests/test_path.py
Comment (sappelhoff) [inline mne_bids/tests/test_path.py:803]:
```suggestion
def test_cross_talk_fpath(return_bids_test_dir):
```

**Review by sappelhoff — COMMENTED** [review]
API looks good to me, just two minor comments.

File: mne_bids/tests/test_path.py
Comment (hoechenberger) [reply mne_bids/tests/test_path.py:803]: Good catch, heh!

File: mne_bids/tests/test_write.py
Comment (hoechenberger) [reply mne_bids/tests/test_write.py:1631]: Addressed in eb6b3fe

File: mne_bids/tests/test_path.py
Comment (hoechenberger) [reply mne_bids/tests/test_path.py]: Addressed in eb6b3fe

**Review by sappelhoff — APPROVED** [review]

**Comment by hoechenberger** [comment]
Hum, what's up with the CI now?!

**Comment by hoechenberger** [comment]
All tests pass locally for me and the error message(s) we see here in the CI don't make any sense to me…

**Comment by hoechenberger** [comment]
Should hopefully pass now after rebasing after #564 has been merged.

**Comment by hoechenberger** [comment]
The tests that are failing now use the stable bids-validator, which doesn't yet support the changes implemented here – only the master branch of bids-validator currently does. How should we deal with this? @sappelhoff?

**Comment by hoechenberger** [comment]
Ok so I've expanded the build matrix to use the stable and the development version of bids-validator on all platforms. All tests with the development version seems to pass now, so this should be good to merge.

**Comment by agramfort** [comment]
@hoechenberger it would be better to skip test if bids_validator does not have the correct version. Merging a PR with red CI is bad.

**Comment by hoechenberger** [comment]
Ugh now the tests are skipped on all platforms… because the bids-validator version number is not updated on GitHub master (is the same as in the latest stable release.) @sappelhoff is this intentional or do you think sb has just forgotten to update the version number?

File: mne_bids/commands/tests/test_cli.py
Comment (hoechenberger) [reply mne_bids/commands/tests/test_cli.py:207]: Would you be okay with using "crosstalk" everywhere in the code? I like this much more than "cross-talk" or "cross_talk".

@sappelhoff?

Comment (hoechenberger) [reply mne_bids/commands/tests/test_cli.py:207]: I've just renamed it.

**Comment by hoechenberger** [comment]
> Ugh now the tests are skipped on all platforms… because the bids-validator version number is not updated on GitHub master (is the same as in the latest stable release.) @sappelhoff is this intentional or do you think sb has just forgotten to update the version number?

I've opened an issue upstream, https://github.com/bids-standard/bids-validator/issues/1080

**Comment by hoechenberger** [comment]
bids-validator has been updated, and I've just restarted the CI jobs here and ensured that we're actually running the all tests (and not accidentally skipping some). All good, all green. Good to merge.

**Review by agramfort — APPROVED** [review]

**Comment by sappelhoff** [comment]
> @sappelhoff is this intentional or do you think sb has just forgotten to update the version number?

this is/was a known issue for a long time --> https://github.com/bids-standard/bids-validator/issues/803

nobody got around to fixing it, and I am not sure whether your PR will persist, because I have no idea about the release process of bids-validator and what kinds of automations are in place that will overwrite/undo your changes - or break because of them :slightly_smiling_face: 

I pinged Ross (the maintainer) for that issue ... best case scenario, your simple fix was exactly what was needed --> let's hope for that.

**Pull request merged**
